molgenis-data-mapper/src/main/java/org/molgenis/data/mapper/controller/MappingServiceController.java

```diff
@@ -496,6 +496,17 @@ public String viewAttributeMapping(@RequestParam(required = true) String mapping
 			attributeMapping = entityMapping.addAttributeMapping(targetAttribute);
 		}
 
+		FieldType targetAttributeDataType = dataService.getEntityMetaData(target).getAttribute(targetAttribute)
+				.getDataType();
+
+		if (targetAttributeDataType instanceof XrefField)
```

Comment:
This can all be done in the ftl because we have targetAttributeMetaData in the model